### DISCUSSION TITLE
Key path cannot refer to static member '....' Error

### DISCUSSION THREAD

**Discussion by artem-krachulov-paysera** [pointfreeco/swift-composable-architecture#3042] [Q&A]
### Description

When I tried to leverage all new features of version 1.7, I faced an error with observing case the path for the nested state. Accessing the case path scoping store shows: `Key path cannot refer to static member '....'`.

<img width="1056" alt="Screenshot 2024-05-02 at 12 09 45 PM" src="https://github.com/pointfreeco/swift-composable-architecture/assets/7997760/60d57a2d-143c-4380-aa03-684fb953c510">

Here is example implemented in both versions:

```
@Reducer
struct StructA {
  @ObservableState
  struct State {
    var structB: StructB
  }
  enum Action {
    case contentA(ContentAReducer.Action)
  }
}

struct StructB {
  var content: Content
  @ObservableState
  public enum Content {
    case contentA(ContentAReducer.State)
  }
}

@Reducer
struct ContentAReducer {
  @ObservableState
  struct State {
    var text: String
  }
  enum Action {}
}

struct ContentAView: View {
  var store: StoreOf<ContentAReducer>
  var body: some View {
    Text(store.text)
  }
}

struct StructAView: View {

  @Binding var store: Store<StructB.Content, StructA.Action>

  var body: some View {

    // Old version works, but soft-deprecated

    SwitchStore(self.store) { content in
      switch content {
      case .contentA:
        CaseLet(/StructB.Content.contentA, action: StructA.Action.contentA) { store in
          ContentAView(store: store)
        }
      }
    }

    // New version prints an error:
    // Key path cannot refer to static member 'contentA'

    switch store.state {
    case .contentA:
      if let store = store.scope(state: \.contentA, action: \.contentA) {
        ContentAView(store: store)
      }
    }
  }
}
```

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### The Composable Architecture version information

1.10.2

### Destination operating system

iOS15

### Xcode version information

Version 15.2 (15C500b)

### Swift Compiler version information

```shell
swift-driver version: 1.87.3 Apple Swift version 5.9.2 (swiftlang-5.9.2.2.56 clang-1500.1.0.2.5)
Target: x86_64-apple-macosx14.0
```

**Comment by stephencelis** [comment]
@artemkrachulov Case key path syntax only works with enums that have the `@CasePathable` macro applied. Further, scoping to enum case state only works with enums that have the `@dynamicMemberLookup` attribute applied.

So you need to add those to your enum and it should work:

```diff
 struct StructB {
   var content: Content
+  @CasePathable
+  @dynamicMemberLookup
   @ObservableState
   public enum Content {
     case contentA(ContentAReducer.State)
   }
 }
```

Note that `@Reducer` automatically applies these attributes to `State` enums, and automatically applies `@CasePathable` to `Action` enums.

See the documentation for more info: https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/reducers#CasePathable-and-dynamicMemberLookup-enums

Since this isn't a bug in the library I'm going to convert to a discussion.

**Reply by artem-krachulov-paysera** [reply]
@stephencelis thank you!
